scrunch/datasets.py

```diff
@@ -2371,15 +2371,23 @@ def fork(self, description=None, name=None, is_published=False,
             dataset otherwise the owner will be the current user in the
             session and the Dataset will be set under `Personal Project`.
         :param project: str, default=None
-           The project ID or URL for the project in which the fork 
-           dataset should be created.
+            The project ID or URL for the project in which the fork
+            dataset should be created.
 
         :returns _fork: scrunch.datasets.BaseDataset
         """
+
         from scrunch.mutable_dataset import MutableDataset
 
         description = description or self.resource.body.description
 
+        LOG.warning(
+            "The preserve_owner parameter will be removed soon"
+            " in favour of providing a project or not "
```

Comment:
A `project` will always be required as far the API is concerned.

Reply:
The idea he is, as an abstraction over the API, the user can either give a project or not, and if they don't, then the fork will be created in the same location as the dataset being forked. What happens below the signature of the method isn't something we want users to think about. In this way, it will be possible for a user to say `ds.fork(..., project=None)`, even though that's not what is implemented internally.

Reply:
if you want to keep the `preserve_owner` abstraction you can future proof it by sending `resource.body.owner` as the `project` in the payload.

Reply:
Thanks, yep that's what happening here:

https://github.com/Crunch-io/scrunch/blob/930905a298dc92b9f02497b59f1da1de508dd934/scrunch/datasets.py#L2420-L2427